lib/ui/painting/image_encoding.cc

```diff
@@ -57,12 +57,22 @@ void InvokeDataCallback(std::unique_ptr<DartPersistentValue> callback,
 
 sk_sp<SkImage> ConvertToRasterImageIfNecessary(sk_sp<SkImage> image,
                                                GrContext* context) {
-  if (context == nullptr) {
-    // The context was null (software rendering contexts) so the image is likely
-    // already a raster image. Nothing more to do.
+  SkPixmap pixmap;
+  if (image->peekPixels(&pixmap)) {
+    // This is already a raster image.
     return image;
   }
 
+  sk_sp<SkImage> raster_image = image->makeRasterImage();
+  if (raster_image) {
+    // The image can be converted to a raster image.
+    return raster_image;
+  }
+
+  if (context == nullptr) {
+    return nullptr;
+  }
+
```

Comment:
Can you add a comment here that makes it clear why `makeRasterImage` would fail. Something like "SkImage::makeRasterImage fails for textures created in one context and transferred to another using SkImage::MakeCrossContextFrom..." would be great.

Reply:
done